.claude/scripts/portfolio-surveyor.test.sh

```
@@ -1224,8 +1224,110 @@ grep -Fq 'dynamic/github-code-scanning/' "${surveyor}" ||
+  fail "surveyor's streak walk must group by run name, not workflow_id alone (#2704)"
+
+# 🔴 The RAW name is not that unit — Dependabot appends a per-run id ("… - Update #1510869626"), so
+# ksail's 48 dynamic/dependabot/ runs on main carry 48 DISTINCT names. An exact-name filter returns
+# one run, the streak can never reach 2, and (REPEATED — ACTIONABLE) can never fire — which would
+# make the exemption permanent for every managed path instead of one run deep, destroying the safety
+# net that justifies the property test. Require the id to be stripped before grouping, and pin the
+# absence of the exact-match form so it cannot come back.
+grep -Fq 'sub("( - Update)? #[0-9]+$"; "")' "${surveyor}" ||
+  fail "surveyor's streak walk must strip the per-run id before grouping, or every streak caps at 1 (#2704)"
+
+grep -Fq 'select(.name == $ENV.RUN_NAME)' "${surveyor}" &&
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)